snuba/clickhouse/translators/snuba/mappers.py

```diff
@@ -25,93 +26,127 @@
 
 
 @dataclass(frozen=True)
-class ColumnToColumn(ColumnMapper):
+class ColumnToExpression(ColumnMapper, ABC):
     """
-    Maps a column with a name and a table into a column with a different name and table.
-
-    The alias is not transformed.
+    Provides some common logic for all the mappers that transform a specific
+    column identified by table and name into one of the allowed expressions
+    a column can be translated into.
+    This exists for convenience just to avoid some code duplication.
     """
 
     from_table_name: Optional[str]
     from_col_name: str
-    to_table_name: Optional[str]
-    to_col_name: str
 
     def attempt_map(
         self,
         expression: ColumnExpr,
         children_translator: SnubaClickhouseStrictTranslator,
-    ) -> Optional[ColumnExpr]:
+    ) -> Optional[
+        Union[ColumnExpr, LiteralExpr, FunctionCallExpr, CurriedFunctionCall]
+    ]:
         if (
             expression.column_name == self.from_col_name
             and expression.table_name == self.from_table_name
         ):
-            return ColumnExpr(
-                alias=expression.alias
-                or qualified_column(self.from_col_name, self.from_table_name or ""),
-                table_name=self.to_table_name,
-                column_name=self.to_col_name,
-            )
+            return self._produce_output(expression, children_translator)
         else:
             return None
 
+    @abstractmethod
+    def _produce_output(
+        self,
+        expression: ColumnExpr,
+        children_translator: SnubaClickhouseStrictTranslator,
```

Comment:
It seems like `children_translator` is not being used in any of th e`_produce_output` methods. Is there a reason we still want to pass it here?

Reply:
It seems like this only exists for the DefaultCurriedFunctionMapper, is that right?

Reply:
You are right. It happens not to be used because all the mappers that inherit from this base class happen to create static expressions from scratch. It is a bit of a coincidence, though I removed it. Should it ever become useful it will be an easy refactoring

Reply:
About DefaultCurriedFunctionMapper, no, it actually has a bigger purpose (See it is used in subscriptableMapper as well). Any rule that translates an expression that is not a simple one (like column) should translate itself and delegate to the children_translator everything else. We just did not encounter such cases yet. They will be more common with multi-table storages etc.